### PR TITLE
Issue 114 compress

### PR DESCRIPTION
**Pull request by eriksiegel**
Attempt to describe the `p:compress` for #114. If this is ok I'll create a proposal for `p:uncompress` as well.

Please niotice what I wrote about the preservation of the document properties. This seemed to me the logical thing to do here but maybe we don't agree on that?

### CODE REVIEW

File: steps/src/main/xml/steps/compress.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  <p:declare-step type="p:compress">
+    <p:input port="source" primary="true" content-types="any" sequence="false"/>
+    <p:output port="result" primary="true" content-types="any" sequence="false"/>
+    <p:option name="format" as="xs:QName" select="'gzip'"/>
+    <p:option name="serialization" as="map(xs:QName,item()*)?"/>
+    <p:option name="parameters" as="map(xs:Qname, item()*)?"/>
+  </p:declare-step>
+
+  <para>The <code>p:compress</code> step first serializes the document appearing on its
```
Comment (gimsieke) [inline steps/src/main/xml/steps/compress.xml]: Please have a look how step names are tagged at other places in the spec. I think it’s `<tag>p:compress</tag>` but I’m not quite sure.

File: steps/src/main/xml/steps/compress.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  <title>p:compress</title>
+
+  <para>The <code>p:compress</code> step serializes the document appearing on its
+      <code>source</code> port and outputs a compressed version of this on its <port>result</port>
+    port.</para>
+
+  <p:declare-step type="p:compress">
+    <p:input port="source" primary="true" content-types="any" sequence="false"/>
+    <p:output port="result" primary="true" content-types="any" sequence="false"/>
+    <p:option name="format" as="xs:QName" select="'gzip'"/>
```
Comment (gimsieke) [inline steps/src/main/xml/steps/compress.xml]: Don’t have the time right now to check – are we using strings as default values for `xs:QName` in other places? I mean, we have the rule about treating uncolonized strings as QNames, but can we use it sloppily for default values? Or do we need to write `select="xs:QName('gzip')"` for the default value?

File: steps/src/main/xml/steps/compress.xml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    <p:option name="serialization" as="map(xs:QName,item()*)?"/>
+    <p:option name="parameters" as="map(xs:Qname, item()*)?"/>
+  </p:declare-step>
+
+  <para>The <code>p:compress</code> step first serializes the document appearing on its
+      <code>source</code>. It then compresses the outcome of this serialization and outputs the
```
Comment (gimsieke) [inline steps/src/main/xml/steps/compress.xml]: `<port>source</port> port` (that is, other tag and add the word „port“). Also please use `<port>` and `<tag>` (or whatever is canonical) below

File: steps/src/main/xml/steps/compress.xml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term><option>serialization</option></term>
+      <listitem>
+        <para>The <option>serialization</option> option is provided to control the serialization of
+          content before compression takes place. If the document to be stored has a
+            <code>serialization</code> property, the serialization is controlled by the merger of
+          the two maps where the entries in the <code>serialization</code> property take precedence.
+          Serialization is described in <biblioref linkend="xproc30"/>.</para>
```
Comment (gimsieke) [inline steps/src/main/xml/steps/compress.xml]: Can we link to a specific section of the XProc 3.0 spec? I think it’s quite rude if we “arbitrarily” split the spec up into multiple parts and thereby lose the ability to link directly to sections of the main spec.

File: steps/src/main/xml/steps/compress.xml (hunk 2 above)
Comment (eriksiegel) [reply steps/src/main/xml/steps/compress.xml]: > are we using strings as default values for xs:QName in other places

Yes, absolutely. You probably didn't notice (?), but most of the text/spec for p:compress was copy/pasted and adapted.

File: steps/src/main/xml/steps/compress.xml (hunk 1 above)
Comment (eriksiegel) [reply steps/src/main/xml/steps/compress.xml]: A quick search shows that both `<code>` and `<tag>` are used. So its not consistent. But I changed it to `<tag>`.

File: steps/src/main/xml/steps/compress.xml (hunk 3 above)
Comment (eriksiegel) [reply steps/src/main/xml/steps/compress.xml]: DOne.

File: steps/src/main/xml/steps/compress.xml (hunk 4 above)
Comment (eriksiegel) [reply steps/src/main/xml/steps/compress.xml]: No, we can't. This is a biblioref and it links to the full resource.

Anyway, this was also copy/paste text and it happens in more places. So if you think this is important, maybe we should open an issue with "finishing touches" or something like that and gather all editorial things that we need/want to do before finally publishing?

**Comment by eriksiegel** [comment]
@ndw @xml-project @gimsieke Could we please either discuss something about it or approve this PR?

**Comment by xml-project** [comment]
@eriksiegel Sorry Erik, I was not aware of this PR. My interface says you ask only @gimsieke for approval.
Is there a way to look at your final PR? If so, I will have a look at it tomorrow afternoon.

**Comment by eriksiegel** [comment]
@xml-project Weird, spec authors is also listed as reviewer, and that's us, right? What goes wrong here?

As always I have no github pages available, but its a single file describing `p:compress` so that shouldn't be too hard.

**Pull request merged**